### ISSUE TITLE
NullRef in ReplaceMethodWithPropertyCodeRefactoringProvider in VB

### ISSUE THREAD

**Issue by dpoeschl** [dotnet/roslyn#17368]
No clue how to repro, but I got the failure info bar on the following code:

```VB
Class C
    Sub Method(a As Integer, b As Integer)
    End Sub
End Class
```

```
System.NullReferenceException : Object reference not set to an instance of an object.
   at Microsoft.CodeAnalysis.VisualBasic.CodeRefactorings.ReplaceMethodWithProperty.VisualBasicReplaceMethodWithPropertyService.GetMethodDeclaration(SyntaxToken token)
   at async Microsoft.CodeAnalysis.ReplaceMethodWithProperty.ReplaceMethodWithPropertyCodeRefactoringProvider.ComputeRefactoringsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService.GetRefactoringFromProviderAsync(<Unknown Parameters>)
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
```

See also: https://developercommunity.visualstudio.com/content/problem/25196/replacemethodwithpropertycoderefactoringprovider-e.html

**Comment by Pilchie** [comment]
Any ideas here?

**Comment by KlausLoeffelmann** [comment]
Got this a couple of times now, when playing around with the new Exception Dialog in VB.
Probably coincidence. 

StackTrace is the same.

**Comment by sophistson** [comment]
Thats closed, but I cant find any solution.
btw: A workaround might be: Just Close VS. That helped me.

**Comment by MarcusMM2** [comment]
still crashes in vs 2017 version 15.2 (26430.13)

**Comment by MarcusMM2** [comment]
public function Read_xxx as integer
 xxx
end sub

-> crash when clicking read_xxx

**Comment by MarcusMM2** [comment]
end function    of course

**Comment by sophistson** [comment]
Be aware of : public function Read_xxx**# ()** as integer

**Comment by wjvander** [comment]
Can confirm crash still happens in Visual Studio 2017 (Pro) v15.2 (_26430.16_)

Stacktrace is the same as mentioned above.

Easy to repoduce, just type `Public Sub` as if you are going to create a method but wait a second or two without typing.

@MarcusMM2 's error does not occur for me

**Comment by CyrusNajmabadi** [comment]
The fix was checked into the 15.3 branch.  If you are on 15.2 you will still hit this issue.
